esm/src/api/services/demo.service.js

```diff
@@ -25,12 +25,17 @@ export const deleteDemoItemService = async (paramsId) => {
 };
 
 export const updateOneDemoItemPropertyValueService = async (paramsId, requestBody) => {
-  const updateOps = {};
-  for (const ops of requestBody) {
-    updateOps[ops.propName] = ops.value;
-  }
-  const query = await Demo.updateOne({ _id: paramsId }, { $set: updateOps }).exec();
-  return query;
+  const queryDoc = await Demo.findById(paramsId).select('_id name age').exec();
+
+  Object.keys(requestBody).forEach(key => {
+    if(key in queryDoc)
+      queryDoc[key] = requestBody[key];
+    else 
+      throw new Error(`property '${key}' does not exist in the mongoose model`)
+  });
+
+  const save = await queryDoc.save();
+  return save;
```

Comment:
**Review comment 5:** Although I'm yet to test this for the 3 templates, but here is a review comment for all 3 templates.
We already are able to access the id of the particular demo element via the `paramsId` argument/parameter from the function.

Which helped mongoose update it at:
````
const query = await Demo.updateOne({ _id: paramsId }, { $set: updateOps }).exec();
````
Meaning we were already able to update by Id. So there is no need to be finding by Id again with `.findById(paramsId)`.

It's preferable to use/retain mongoose `.updateOne()` method, that way it's even clear that we are "updating" something that is already existing.

#

You may have been able to do your check and throw errors just after the previously existing for-of loop that was there before.

There is no need to modify the code in the function this much.